src/main/java/com/trivialTrivia/persistence/TriviApiDao.java

```diff
@@ -0,0 +1,92 @@
+package com.trivialTrivia.persistence;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+import org.hibernate.Session;
+
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+import java.util.List;
+
+/**
+ * The type Trivi api dao.
+ *
+ * @param <T> the type parameter
+ */
+public class TriviApiDao<T> {
```

Comment:
I don't believe you'll need to take type anywhere in this dao since we're only using it for type Trivia